### PR TITLE
extensions/ext: Add VK_EXT_mesh_shader

### PR DESCRIPTION
**Pull request by BeastLe9enD**
Adds support for `VK_EXT_mesh_shader`.

### CODE REVIEW

**Comment by MarijnS95** [comment]
Changelog?

File: ash/src/extensions/ext/mesh_shader.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        );
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdDrawMeshTasksIndirectEXT.html>
+    #[inline]
+    pub unsafe fn cmd_draw_mesh_tasks_indirect(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        buffer: vk::Buffer,
+        offset: vk::DeviceSize,
```
Comment (MarijnS95) [inline ash/src/extensions/ext/mesh_shader.rs:43]: Should we document a reference to https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkDrawMeshTasksIndirectCommandEXT.html for the type that constitutes `Buffer`?

Comment (BeastLe9enD) [reply ash/src/extensions/ext/mesh_shader.rs:43]: Would be an option. However, it is not present for other indirect functions at the moment.

Comment (MarijnS95) [reply ash/src/extensions/ext/mesh_shader.rs:43]: I did it for RT pipelines:

https://github.com/ash-rs/ash/blob/f52ad9f08e49da51e8f4662e7ea800e6a493f98a/ash/src/extensions/khr/ray_tracing_pipeline.rs#L131

Comment (BeastLe9enD) [reply ash/src/extensions/ext/mesh_shader.rs:43]: Yes maybe its better to just start adding it to mesh shaders too, I'm very happy to do it if its wanted!

Comment (MarijnS95) [reply ash/src/extensions/ext/mesh_shader.rs:43]: Yeah, would be nice. We have the types so might as well set up an intradoc link to them.

File: ash/src/extensions/ext/mesh_shader.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            command_buffer,
+            group_count_x,
+            group_count_y,
+            group_count_z,
+        );
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdDrawMeshTasksIndirectEXT.html>
+    ///
+    /// `buffer` is a buffer which contains [`vk::DrawMeshTasksIndirectCommandEXT`] structures containing the draw parameters.
```
Comment (MarijnS95) [inline ash/src/extensions/ext/mesh_shader.rs:39]: Nit: not a big fan of the `contains ... containing` repetition, can we word this differently?

Any other parameters we can mention? For example:

```suggestion
    /// `buffer` contains `draw_count` [`vk::DrawMeshTasksIndirectCommandEXT`] structures starting at `offset` in bytes, holding the draw parameters.
```

Similar below.

Comment (BeastLe9enD) [reply ash/src/extensions/ext/mesh_shader.rs]: Sure :)

File: ash/src/extensions/ext/mesh_shader.rs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    #[inline]
+    pub unsafe fn cmd_draw_mesh_tasks_indirect_count(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        buffer: vk::Buffer,
+        offset: vk::DeviceSize,
+        count_buffer: vk::Buffer,
+        count_buffer_offset: vk::DeviceSize,
+        max_draw_count: u32,
+        stride: u32,
```
Comment (MarijnS95) [inline ash/src/extensions/ext/mesh_shader.rs:70]: As mentioned (_similar below_, not the same) this comment has to be adjusted for the parameters here. Sorry for yet another nit before merging this in, but I thought this was clear :wink:

Comment (BeastLe9enD) [reply ash/src/extensions/ext/mesh_shader.rs]: Sorry for the unnecessary error, I was doing 3 things at the same time again and didn't think along :)

**Review by MarijnS95 — APPROVED** [review]

File: ash/src/extensions/ext/mesh_shader.rs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            offset,
+            draw_count,
+            stride,
+        );
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdDrawMeshTasksIndirectCountEXT.html>
+    ///
+    /// `buffer` contains a maximum of `max_draw_count` [`vk::DrawMeshTasksIndirectCommandEXT`] structures starting at `offset` in bytes, holding the draw parameters.
+    /// `count_buffer` is the buffer containing the draw count, starting at `count_buffer_offset` in bytes. The actual number of executed draw calls is the minimum of the count specified in `count_buffer` and `max_draw_count`.
```
Comment (MarijnS95) [inline ash/src/extensions/ext/mesh_shader.rs:61]:
```suggestion
    /// `count_buffer` is the buffer containing the draw count, starting at `count_buffer_offset` in bytes.
    /// The actual number of executed draw calls is the minimum of the count specified in `count_buffer` and `max_draw_count`.
```

Comment (MarijnS95) [reply ash/src/extensions/ext/mesh_shader.rs]: Seems like you disabled contributions by maintainers, I cannot apply this suggestion to your branch.

**Pull request merged**